nexus/src/app/sagas/disk_delete.rs

```diff
@@ -188,6 +196,96 @@ async fn sdd_account_space_undo(
     Ok(())
 }
 
+async fn sdd_deallocate_local_storage(
+    sagactx: NexusActionContext,
+) -> Result<(), ActionError> {
+    let osagactx = sagactx.user_data();
+    let params = sagactx.saga_params::<Params>()?;
+    let opctx = crate::context::op_context_for_saga_action(
+        &sagactx,
+        &params.serialized_authn,
+    );
+
+    let datastore::Disk::LocalStorage(disk) = params.disk else {
+        // Check during make_saga_dag prevents this case.
+        unreachable!();
```

Comment:
nit, take it or leave it: why not
```suggestion
        unreachable!(
            "the check during `make_saga_dag` should ensure the disk type is local storage",
        );
```
or something?

Reply:
that's definitely better: 404665592195936148ad0041d0d01df380d24b09